src/modules/alsa/module-alsa-card.c

```diff
@@ -708,12 +708,13 @@ static pa_hook_result_t source_output_unlink_hook_callback(pa_core *c, pa_source
     return PA_HOOK_OK;
 }
 
-static pa_card_profile *find_best_profile_with_available_ports(pa_card *card, pa_direction_t direction) {
+static pa_card_profile *find_best_available_profile(pa_card *card, pa_direction_t direction) {
```

Comment:
I don't fully grasp the commit message here. Are you saying:

With this fix one can use module-switch-on-connect to automatically move
streams to a HDMI port on a separate card (which is the current
situation with AMD graphics) [when a HDMI TV is connected to the laptop] and have the streams moved back by
module-rescue-stream [when the HDMI TV becomes disconnected].

Reply:
More or less -- this specific fix only takes care of making sure the card does not get stuck in a profile with a non-working sink if nothing else changes the card profile first (for example, module-switch-on-port-available), so the other modules you mentioned can do their jobs. I'll try to make the commit message here more clear.